sssd_test_framework/utils/local_override.py

```diff
@@ -0,0 +1,414 @@
+from __future__ import annotations
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.cli import CLIBuilder, CLIBuilderArgs
+
+__all__ = [
+    "LocalOverrideUtils",
+    "LocalOverrideUser",
+    "LocalOverrideGroup",
+]
+
+
+class LocalOverrideUtils(MultihostUtility[MultihostHost]):
+    """
+    Management of local override users and groups, i.e. sss_override.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+
+        self.cli: CLIBuilder = CLIBuilder(host.ssh)
+        self._users: list[str] = []
+        self._groups: list[str] = []
+
+    def teardown(self) -> None:
```

Comment:
I was thinking about this and it might be better to avoid this and rather change SSSDUtils like this:

```diff
diff --git a/sssd_test_framework/utils/sssd.py b/sssd_test_framework/utils/sssd.py
index 6e14029..adfd943 100644
--- a/sssd_test_framework/utils/sssd.py
+++ b/sssd_test_framework/utils/sssd.py
@@ -110,6 +110,11 @@ class SSSDUtils(MultihostUtility[MultihostHost]):
         )
         self.svc.reload_daemon()
 
+        # Backup existing SSSD data
+        self.fs.backup("/etc/sssd")
+        self.fs.backup("/var/lib/sss")
+        self.fs.backup("/var/log/sssd")
+
         if self.__load_config:
             self.config_load()
             return
```

This will make sure that all changes in SSSD data is automatically reverted regardless of any tool that we use and we do not have to deal with it individually. And just document it here.

Reply:
I wouldn't name it reload, but reset. It's confusing because reload when talking system services just means refreshing the configuration without needing to restart the daemon.

Reply:
Yes, Just a thought.